protocol/directory.go

```diff
@@ -110,6 +110,12 @@ func (d *ConiksDirectory) Register(req *RegistrationRequest) (
 		d.tbs[req.Username] = tb
 		return NewRegistrationProof(ap, d.LatestSTR(), tb, Success)
 	} else {
+		currentAP, _ := d.pad.LookupInCurrentEpoch(req.Username)
+		if bytes.Equal(currentAP.LookupIndex, currentAP.Leaf.Index) {
+			// use the ap from the latest STR to return to the client
+			return NewRegistrationProof(ap, d.LatestSTR(), nil, ErrorNameExisted)
```

Comment:
My only issue with this is that the client doesn't get any proof that the name already exists, and until the current tree is committed, I don't know if we should be relying on it to give us information since a malicious server can now easily cheat and give us false information about which names exist, don't exist etc and we wouldn't have any proof. I'm starting to think that TBs shouldn't be just an extension :( At least while we don't have a better solution for signed promises.